src/mono/browser/runtime/loader/config.ts

```diff
@@ -283,30 +286,53 @@ export function isDebuggingSupported (): boolean {
 async function loadBootConfig (module: DotnetModuleInternal): Promise<void> {
     const defaultConfigSrc = loaderHelpers.locateFile(module.configSrc!);
 
-    const loaderResponse = loaderHelpers.loadBootResource !== undefined ?
-        loaderHelpers.loadBootResource("manifest", "blazor.boot.json", defaultConfigSrc, "", "manifest") :
-        defaultLoadBootConfig(defaultConfigSrc);
+    let loaderResponse = null;
+    if (loaderHelpers.loadBootResource !== undefined) {
+        loaderResponse = loaderHelpers.loadBootResource("manifest", "blazor.boot.json", defaultConfigSrc, "", "manifest");
+    }
 
     let loadConfigResponse: Response;
 
     if (!loaderResponse) {
-        loadConfigResponse = await defaultLoadBootConfig(appendUniqueQuery(defaultConfigSrc, "manifest"));
+        if (defaultConfigSrc.includes(".json")) {
+            loadConfigResponse = await fetchBootConfig(appendUniqueQuery(defaultConfigSrc, "manifest"));
+        } else {
+            const loadedConfig = await importBootConfig(appendUniqueQuery(defaultConfigSrc, "manifest"));
+            deep_merge_config(loaderHelpers.config, loadedConfig);
+            return;
+        }
     } else if (typeof loaderResponse === "string") {
-        loadConfigResponse = await defaultLoadBootConfig(makeURLAbsoluteWithApplicationBase(loaderResponse));
+        if (loaderResponse.includes(".json")) {
+            loadConfigResponse = await fetchBootConfig(makeURLAbsoluteWithApplicationBase(loaderResponse));
+        } else {
+            const loadedConfig = await importBootConfig(makeURLAbsoluteWithApplicationBase(loaderResponse));
+            deep_merge_config(loaderHelpers.config, loadedConfig);
+            return;
+        }
     } else {
         loadConfigResponse = await loaderResponse;
+        if (typeof loadConfigResponse.json != "function") {
+            // If the response doesn't contain .json(), consider it an imported module.
+            deep_merge_config(loaderHelpers.config, (loadConfigResponse as any).config);
```

Comment:
`deep_merge_config` could be the same in all branches, right ? Could we move it down ?